New parallel python tutotials (#744)
NREL/HPC: docs/Documentation/Development/Languages/Python/KestrelParallelPytonJupyter/README.md

> # Interactive Parallel Python on Kestrel with Jupyter

Reviewer: Is this page supposed to be rendered in the github pages site, or is it intended to go here: https://github.com/NREL/HPC/tree/master/general/Jupyterhub?

Author: Hi Haley, I intend to keep it here for the time being. The main Jupyterhub page has many references to Eagle, which is going to need a lot of editing. I will be doing that next and then try and combine the content of both of these pages. Thanks.

> For a general introduction to using Jupyter notebooks on Kestrel, please refer to the [official documentation](https://nrel.github.io/HPC/Documentation/Development/Jupyter/). For quick low compute intensive jobs, please visit the [Kestrel JupyterHub](https://kestrel-jhub.hpc.nrel.gov/). This page conveys specific information on how to leverage parallel computing on NREL HPC systems with python through Jupyter notebooks. Accompanying notebooks to test if the environment and the allocation has been configured correctly are in the 'exampleNotebooks' folder.

Reviewer: ```suggestion
For a general introduction to using Jupyter notebooks on Kestrel, please refer to the [official documentation](https://nrel.github.io/HPC/Documentation/Development/Jupyter/). This page covers how to leverage parallel computing with python through Jupyter notebooks on compute nodes. Accompanying notebooks to test if the environment and the job have been configured correctly are in the 'exampleNotebooks' folder.
```

Reviewer: kestrel jupyterhub is covered on the documentation page that you linked to and isn't really relevant to this page so I don't think it needs to be mentioned here to make the paragraph more concise.

## Install packages

CuPy : [“An open-source array library for GPU-accelerated computing with Python”](https://cupy.dev/)
```
$ nvcc –version
$ conda install -c conda-forge cupy
$ conda install -c conda-forge cupy cudnn cutensor nccl
```

numba-cuda : [“CUDA GPU programming by directly compiling a restricted subset of Python code into CUDA kernels”](https://nvidia.github.io/numba-cuda/user/index.html)
```
$ conda install -c conda-forge numba-cuda
$ conda install -c conda-forge cuda-nvcc cuda-nvrtc "cuda-version>=12.0"
```

mpi4py : [“Python bindings for the Message Passing Interface (MPI) standard to exploit multiple processors”](https://mpi4py.readthedocs.io/en/stable/) 
```
$ conda install -c conda-forge mpi4py openmpi
$ conda install cuda-cudart cuda-version=12
```

ipyparallel : [“Interactive Parallel Computing with IPython”](https://ipyparallel.readthedocs.io/en/latest/)
```
$ conda install ipyparallel
```

Dask : [“A flexible open-source Python library for parallel and distributed computing”](https://www.dask.org/)
```
$ conda install dask
$ conda install dask-jobqueue
$ conda install graphviz
$ conda install ipycytoscape
$ conda install matplotlib
```

## Launching jobs

> A general guide to allocating resources for running jobs on the Kestrel can be found in the [official documentation](https://nrel.github.io/HPC/Documentation/Systems/Kestrel/Running/). Below are example proceedures suitable for running jobs involving specific python modules, depending on their parallelisation capability. 

Reviewer: ```suggestion
A general guide to running jobs on Kestrel can be found in the [official documentation](https://nrel.github.io/HPC/Documentation/Systems/Kestrel/Running/). Below are example procedures suitable for running jobs involving specific python modules, depending on their parallelization capability. 
```

> The text in red box shows an example of the output parameter `<nodename>` and the text in yellow box shows an example of the output parameter `<alphabet soup>`, relevant to the following tutorial.

Reviewer: ```suggestion
The text in the red box shows an example of the output parameter `<nodename>` and the text in the yellow box shows an example of the output parameter `<alphabet soup>`, relevant to the following tutorial.
```
